citar.el

```diff
@@ -277,6 +277,9 @@ of all citations in the current buffer."
   :group 'citar
   :type 'alist)
 
+(defvar citar-major-modes '(org-mode latex-mode markdown-mode)
```

Comment:
What about text-mode? And would that cover emails?

Reply:
The variable is used in interactive specification of `citar-insert-citation`, to prevent the issue raised in #441. So it only lists modes in which citations can be inserted.

Thoughts on a different name and/or docstring?

Reply:
In other words, it wouldn't have any bearing on the use of Citar in text-mode or in emails.

I gave it such a general name since it is just a list of modes derived from 'citar-major-mode-functions'. Perhaps it would make sense to mention that in the docstring?

Something like:
"List of major modes for which there is major mode specific functionality, per 'citar-major-mode-functions'."

The other option --- which I suggested was overkill, but maybe isn't --- would be to skip adding this defvar and make the list programmatically in 'citar-insert-citation' itself, based on `citar-major-mode-function`.

Reply:
I'm unclear why we need the defvar to begin with. The relevant function only works if there's a corresponding major-mode function.

Right?

So what am I missing?

Reply:
https://github.com/bdarcus/citar/issues/441#issuecomment-979170594

It's to avoid the UX issue mentioned here. Currently, the user can call `insert-citation` in an unsupported mode and go through with selecting a reference, only to be told afterwards that citation-insertion is not allowed in that mode.